Document/0x05g-Testing-Network-Communication.md

```diff
@@ -117,6 +117,14 @@ Certificate pinning is the process of associating the backend server with a part
 
 The certificate can be pinned and hardcoded into the app or retrieved at the time the app first connects to the backend. In the latter case, the certificate is associated with ("pinned" to) the host when the host is seen for the first time. This alternative is less secure because attackers intercepting the initial connection can inject their own certificates.
 
+Note that there are various ways to work with a failing pin:
+
+- Inform the user about not being able to connect to the back-end and stop all operations. The app can check whether there is an update and inform the user about updating to the latest version of the app if available. 
+  - Additionally a call can be made to the back-end or to a crash-reporting service that the pinning has failed to inform the developer and/or back-end of the attack or misconfiguration.
```

Comment:
note that below you write: "the back-end or crash-reporting service should be informed..." maybe decide on operation team vs crash-reporting service?

Reply:
I rather do not any of these :). For me, the most important is that the crash-reporting service and/or the backend-teams get them (given devops/secdevops). Based on that the teams can do what they want but i want either:
- the app-developer to see the crash reported so they know they have to fix stuff in case of a security misconfiguration, which is most often the case in terms of volume and user-frustration
- the backend-developer who is monitoring the systems (Devops 👍 )and sees these items comming in
- the risk-engine in case a security event is raised for a given customer, to indicate compromise of teh app if it is only a single app instance on which this happens.

Reply:
Changed the text accordingly